text/0000-cnb-env-var.md

```diff
@@ -0,0 +1,96 @@
+# Meta
+[meta]: #meta
+- Name: Support for env vars in creator
+- Start Date: 2023-09-07
+- Author(s): [kappratiksha](https://github.com/kappratiksha)
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Add a flag for passing the build time env variables to buildpacks when invoking creator
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+When invoking the creator process, there is no way to pass build time environment variables to the buildpacks other than creating the files under /platform/env folder. This adds an extra step for the authors when implementing their buildpack. Adding a new flag to the creator process will simplify the process of setting the environment variables.
+
+We already have something similar for the [pack cli tool](https://buildpacks.io/docs/tools/pack/cli/pack_build/) and it proves to be very convenient. Example:
+
+```
+pack build test_img --env VAR1=VALUE1
+```
+
+# What it is
+[what-it-is]: #what-it-is
+
+A flag that can be used by buildpack user to pass down the environment variables while invoking creator.
+
+/cnb/lifecycle/creator -env stringArray 
```

Comment:
Can you add details on order of precedence? 

E.g. If a platform spits values into `/platform/env` today, say from a `project.toml`. The `project.toml` has `FOO=bar`. The platform decides that they always wanted to set `FOO=baz` via the new `-env`, what happens.